From: SJ Park <sj@kernel.org>
Date: Sat, 15 Aug 2026 10:44:51 -0700
Subject: [PATCH] mm/damon/core: use mult_frac() in
 damon_feed_loop_next_input()

Since score_goal_diff is already smaller than goal, it is safe from
overflow.

Signed-off-by: SJ Park <sj@kernel.org>
---
 mm/damon/core.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 348630fba70c0..a283e9ae6ff9d 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3094,11 +3094,7 @@ static unsigned long damon_feed_loop_next_input(unsigned long last_input,
 		return min_input;
 
 	score_goal_diff = abs_diff(score, goal);
-
-	if (last_input < ULONG_MAX / score_goal_diff)
-		compensation = last_input * score_goal_diff / goal;
-	else
-		compensation = last_input / goal * score_goal_diff;
+	compensation = mult_frac(last_input, score_goal_diff, goal);
 
 	if (over_achieving)
 		return max(last_input - compensation, min_input);
-- 
2.47.3

